lib/pcgsperm.gi

```diff
@@ -1089,30 +1089,32 @@ local r,i,p,A,f,a;
   # is the group in the mappings families cache?
   f:=FamiliesOfGeneralMappingsAndRanges(FamilyObj(OneOfPcgs(pcgs)));
   i:=1;
-  while i<=Length(f) do
-    a:=ElmWPObj(f,i);
-    if a<>fail and IsBound(a!.DefiningPcgs) 
-       and RelativeOrders(a!.DefiningPcgs)=r then
-      # right type PCGS -- test relations
-      a:=a!.DefiningPcgs;
-      if (needindices=false or  (not HasIndicesEANormalSteps(a)) or
-	IndicesEANormalSteps(a)=IndicesEANormalSteps(pcgs)) and
-	ForAll([1..Length(r)-1],x->
-        ExponentsOfPcElement(a,a[x]^r[x])
-        =ExponentsOfPcElement(pcgs,pcgs[x]^r[x])) and
-	ForAll([1..Length(r)],x->ForAll([x+1..Length(r)],y->
-	  ExponentsOfPcElement(a,a[y]^a[x])
-	  =ExponentsOfPcElement(pcgs,pcgs[y]^pcgs[x]))) then
-
-	# indeed the group is OK
-	if not HasIndicesEANormalSteps(a) then
-	  SetIndicesEANormalSteps(a,IndicesEANormalSteps(pcgs));
-	fi;
-        A:=GroupOfPcgs(a);
-	return A;
+  atomic readonly GENERAL_MAPPING_REGION do
```

Comment:
@markuspf : changing `readonly` to `readwrite` fixed the access problem to a weak pointer object from #26, but I don't understand from the code why `readwrite` is needed. Before that, I was able to reproduce the problem three times in a row.